src/ui/public/visualize/visualization_editor.js

```diff
@@ -13,25 +13,24 @@ uiModules
 
   return {
     restrict: 'E',
-    require: '?renderCounter',
     scope: {
       vis: '=',
       visData: '=',
       uiState: '=?',
-      searchSource: '='
+      searchSource: '=',
+      savedObj: '='
     },
     link: function ($scope, element) {
       // Clone the _vis instance.
       const vis = $scope.vis;
+      const savedObj = $scope.savedObj;
       const Editor = typeof vis.type.editor === 'function' ? vis.type.editor :
         editorTypes.find(editor => editor.key === vis.type.editor);
-      const editor = new Editor(element[0], vis);
+      const editor = new Editor(element[0], vis, savedObj);
```

Comment:
do we still need this `vis` to be passed in separately ? its available on the savedObj

Reply:
Good point, we do not. If we're comfortable passing the savedObject itself all the way here, I can refactor the code to grab `vis` off of `savedObj`.